### PR TITLE
Strategy Scheduler.

### PR DESCRIPTION
**Pull request by MichaelRawson**
This PR implements a scheduler within Vampire to execute strategy sets with. The current `PortfolioMode` is then re-implemented as a degenerate case where the scheduler simply returns schedules in sequential order.

This should likely not be merged without scrutiny - there's a couple of architectural problems at the very least that should be fixed first - see FIXME in `PortfolioMode.hpp`. Plus bugs, probably. Seems to work alright on a few problems though.

### CODE REVIEW

**Comment by selig** [comment]
Let me look through the code.

Can this run with current schedules out of the box? If so, I suggest we run a StarExec job in debug mode across all of TPTP. It's a good initial test.

**Comment by selig** [comment]
@quickbeam123 can you briefly scan through the changes made to Portfolio mode as you changed it recently?

**Comment by MichaelRawson** [comment]
@selig : yes, seems to work fine so far. OK - will look into how to run a StarExec job for this.

Please do look at the changes to portfolio mode, I'm not sure what a lot of it does.

File: Lib/Sys/Multiprocessing.cpp [hunk 1]
```diff
@@ -224,6 +224,24 @@ void Multiprocessing::kill(pid_t child, int signal)
...
+
+pid_t Multiprocessing::poll_children(bool &stopped, bool &exited, int &code)
+{
+  CALL("Multiprocessing::poll_child");
+
+  int status;
+  pid_t pid = waitpid(-1, &status, WUNTRACED);
+  stopped = WIFSTOPPED(status);
+  exited = WIFEXITED(status);
+  if(exited)
```
Comment (selig) [inline Lib/Sys/Multiprocessing.cpp]: Add {} please

**Comment by MichaelRawson** [comment]
So a StarExec comparison has been done, and this branch proves a comparable number (actually slightly more) than baseline Vampire from recent `master`: 12,741 vs 12,727. Analysis of the job data also does not indicate any questionable proofs.

**Pull request merged**